src/credentials/loginManager.ts

```diff
@@ -55,4 +69,45 @@ export class LoginManager {
     public async logout(): Promise<void> {
         await this.awsContext.setCredentials(undefined)
     }
+
+    /**
+     * Updates the CredentialsStore if the credentials are considered different
+     */
+    private async updateCredentialsStore(credentialsId: string, provider: CredentialsProvider): Promise<void> {
+        const currentHash = this.credentialsStore.getCredentialsHashCode(credentialsId)
+        if (provider.getHashCode() !== currentHash) {
```

Comment:
provider `getHashCode` is weird because it's the hash code of the current profile not the hash code of the provider, which made me really confused reading this

Reply:
Agreed. What do you think about treating the cache as one object (that contains credentials and cache, and possibly other props in the future)?

Reply:
I've updated the cache to get/set Credentials + props as an object